hibernate-core/src/test/java/org/hibernate/test/collection/delayedOperation/DetachedBagDelayedOperationTest.java

```diff
@@ -0,0 +1,387 @@
+				}
+		);
+	}
+
+	@Test
+	@TestForIssue( jiraKey = "HHH-11209" )
+	public void testMergeDetachedCollectionWithQueuedOperations() {
+		final Parent pOriginal = doInHibernate(
+				this::sessionFactory, session -> {
+					Parent p = session.get( Parent.class, 1L );
+					assertFalse( Hibernate.isInitialized( p.getChildren() ) );
+					// initialize
+					Hibernate.initialize( p.getChildren() );
+					assertTrue( Hibernate.isInitialized( p.getChildren() ) );
+					return p;
+				}
+		);
+		final Parent pWithQueuedOperations = doInHibernate(
+				this::sessionFactory, session -> {
+					Parent p = (Parent) session.merge( pOriginal );
+					Child c = new Child( "Zeke" );
+					c.setParent( p );
+					session.persist( c );
+					assertFalse( Hibernate.isInitialized( p.getChildren() ) );
+					p.getChildren().add( c );
+					assertFalse( Hibernate.isInitialized( p.getChildren() ) );
+					assertTrue( ( (AbstractPersistentCollection) p.getChildren() ).hasQueuedOperations() );
+
+					checkTriggerablesNotTriggered();
+					session.detach( p );
+					assertTrue( triggerableQueuedOperationWhenDetachFromSession.wasTriggered() );
+					assertEquals(
+							"HHH000496: Detaching an uninitialized collection with queued operations from a session: [org.hibernate.test.collection.delayedOperation.DetachedBagDelayedOperationTest$Parent.children#1]",
+							triggerableQueuedOperationWhenDetachFromSession.triggerMessage()
+					);
+					triggerableQueuedOperationWhenDetachFromSession.reset();
+
+					// Make sure nothing else got triggered
+					checkTriggerablesNotTriggered();
+
+					return p;
+				}
+		);
+
+		checkTriggerablesNotTriggered();
+
+		assertTrue( ( (AbstractPersistentCollection) pWithQueuedOperations.getChildren() ).hasQueuedOperations() );
+
+		// Merge detached Parent with uninitialized collection with queued operations
+		doInHibernate(
+				this::sessionFactory, session -> {
+
+					checkTriggerablesNotTriggered();
+
+					assertFalse( triggerableIgnoreQueuedOperationsOnMerge.wasTriggered() );
+					Parent p = (Parent) session.merge( pWithQueuedOperations );
+					assertTrue( triggerableIgnoreQueuedOperationsOnMerge.wasTriggered() );
+					assertEquals(
+							"HHH000494: Attempt to merge an uninitialized collection with queued operations; queued operations will be ignored: [org.hibernate.test.collection.delayedOperation.DetachedBagDelayedOperationTest$Parent.children#1]",
+							triggerableIgnoreQueuedOperationsOnMerge.triggerMessage()
+					);
+					triggerableIgnoreQueuedOperationsOnMerge.reset();
+
+					assertFalse( Hibernate.isInitialized( p.getChildren() ) );
+					assertFalse( ( (AbstractPersistentCollection) p.getChildren() ).hasQueuedOperations() );
+
+					// When initialized, p.children will not include the new Child ("Zeke"),
+					// because that Child was flushed without a parent before being detached
+					// along with its parent.
+					Hibernate.initialize( p.getChildren() );
```

Comment:
When initialized, the collection does not contain the `Child` that was added to the collection by the queued operations before being detached. That's because the queued operations in the entity being merged (`pWithQueuedOperations`) were ignored.